### PR TITLE
[Tanium] Create two connectors (external import and stream) to replace actual Tanium Stream connector

### PR DESCRIPTION
**Pull request by Powlinett**
### Proposed changes

* Create external-import/tanium-incidents and use external import template to refacto
* Create stream/tanium-intel and use stream template to refacto

### Related issues

* #2298 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality using different use cases
- [ ] I added/updated the relevant documentation (either on github or on notion)
  - only READMEs are updated for now
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Comment by romain-filigran** [comment]
@SamuelHassine : what's your idea for the name to give to new "stream" connector ? tanium-?

File: external-import/tanium-incidents/README.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+# OpenCTI Tanium Incidents Connector
+
+<!--
+General description of the connector
+* What it does
+* How it works
+* Special requirements
+* Use case description
+* ...
+-->
```
Comment (flavienSindou) [inline external-import/tanium-incidents/README.md:10]: I think you could get rid off this comment

File: external-import/tanium-incidents/README.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+| `tanium_incidents_url`               | `TANIUM_INCIDENTS_URL`               | Yes       | The Tanium instance API URL.                                                            |
+| `tanium_incidents_url_console`       | `TANIUM_INCIDENTS_URL_CONSOLE`       | Yes       | The Tanium instance console URL.                                                        |
+| `tanium_incidents_ssl_verify`        | `TANIUM_INCIDENTS_SSL_VERIFY`        | Yes       | Enable the SSL certificate check (default: `true`)                                      |
+| `tanium_incidents_token`             | `TANIUM_INCIDENTS_TOKEN`             | Yes       | The Tanium login user.                                                                  |
+| `tanium_incidents_import_alerts`     | `TANIUM_INCIDENTS_IMPORT_ALERTS`     | No        | Enable alerts import                                                                    |
+| `tanium_incidents_import_start_date` | `TANIUM_INCIDENTS_IMPORT_START_DATE` | No        | Import starting date (in YYYY-MM-DD format) - used only if connector's state is not set |
```
Comment (flavienSindou) [inline external-import/tanium-incidents/README.md]: This seems to be a duplicated in this readme file

File: external-import/tanium-incidents/src/tanium_incidents_connector/config_variables.py
Comment (flavienSindou) [inline external-import/tanium-incidents/src/tanium_incidents_connector/config_variables.py:1]: In accordance with readme, config yaml and docker compose 👍

File: external-import/tanium-incidents/src/tanium_incidents_connector/connector.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+                        stix_objects = self._collect_intelligence(alert)
+                        if stix_objects:
+                            stix_objects.append(self.author)
+                            stix_bundle = self.helper.stix2_create_bundle(stix_objects)
+
+                            friendly_name = (
+                                self.helper.connect_name
+                                + "run @ "
+                                + datetime.now().astimezone(pytz.UTC).isoformat()
+                            )
```
Comment (flavienSindou) [inline external-import/tanium-incidents/src/tanium_incidents_connector/connector.py:292]: I like this idea 👍

**Review by flavienSindou — COMMENTED** [review]
Some comments in readme files.
Otherwise it looks ok to me

File: external-import/tanium-incidents/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+services:
+  connector-tanium:
+    image: opencti/connector-tanium-incidents:6.3.3
+    environment:
+      - OPENCTI_URL=http://localhost
+      - OPENCTI_TOKEN=ChangeMe
+      - CONNECTOR_ID=ChangeMe
+      - "CONNECTOR_NAME=Tanium Sightings"
+      - CONNECTOR_SCOPE=tanium
+      - CONNECTOR_CONFIDENCE_LEVEL=60 # From 0 (Unknown) to 100 (Fully trusted)
```
Comment (helene-nguyen) [inline external-import/tanium-incidents/docker-compose.yml]: You can remove confidence_level as it is deprecated now

File: stream/tanium/README.md [hunk 5]
```diff
@@ -19,21 +24,21 @@ OpenCTI data is coming from *import* connectors. Once this data is ingested in O
...
+| `connector_id`                | `CONNECTOR_ID`                | Yes       | A valid arbitrary `UUIDv4` that must be unique for this connector.                                                         |
+| `connector_live_stream_id`    | `CONNECTOR_LIVE_STREAM_ID`    | Yes       | The Live Stream ID of the stream created in the OpenCTI interface                                                          |
+| `connector_name`              | `CONNECTOR_NAME`              | Yes       | The name of the Tanium instance, to identify it if you have multiple Tanium connectors.                                    |
+| `connector_scope`             | `CONNECTOR_SCOPE`             | Yes       | Must be `tanium`, not used in this connector.                                                                              |
+| `connector_confidence_level`  | `CONNECTOR_CONFIDENCE_LEVEL`  | Yes       | The default confidence level for created sightings (a number between 1 and 4).                                             |
```
Comment (helene-nguyen) [inline stream/tanium/README.md:35]: You can remove confidence_level as it is deprecated now

File: external-import/tanium-incidents/src/tanium_incidents_connector/connector.py (hunk 3 above)
Comment (Powlinett) [reply external-import/tanium-incidents/src/tanium_incidents_connector/connector.py:292]: It's from the template if I'm correct 🙂

File: stream/tanium/README.md (hunk 5 above)
Comment (Powlinett) [reply stream/tanium/README.md:35]: It's still in use in this connector (the former one), so I'd rather not remove it as the whole connector will be deprecated anyway. What do you think?

Comment (helene-nguyen) [reply stream/tanium/README.md:35]: Saw you have removed it :)

**Review by Megafredo — APPROVED** [review]

**Comment by helene-nguyen** [comment]
Everything look good to me, tested locally.
Great job 😊

**Review by helene-nguyen — APPROVED** [review]

**Review by flavienSindou — APPROVED** [review]
LGTM

**Pull request merged**